src/libs/intents/localMethods.ts

```diff
@@ -87,13 +87,19 @@ const openAppOSSettings = async (): Promise<null> => {
 }
 
 export const internalMethods = {
-  setFlagshipUI: (intent: FlagshipUI): Promise<null> =>
-    setFlagshipUI(
-      intent,
-      EnvService.nameIs(strings.environments.test)
-        ? internalMethods.setFlagshipUI.caller?.name // eslint-disable-line @typescript-eslint/no-unnecessary-condition
-        : ''
-    )
+  setFlagshipUI: (intent: FlagshipUI): Promise<null> => {
+    const caller = (): string => {
+      if (!EnvService.nameIs(strings.environments.test)) return 'unknown'
+
+      try {
+        return internalMethods.setFlagshipUI.caller.name
+      } catch (error) {
+        return 'unknown'
```

Comment:
What is the advantage of try...catching this instead of using optional chaining operators?

Reply:
Even with them, it throws an error. I'm not sure exactly what is the problem, but it's linked to security. It doesn't allow me to get the caller name in some contexts